File: java/hamming/src/main/java/Hamming.java
import java.util.stream.IntStream;

class Hamming {
    private int hamming;

    Hamming(String leftStrand, String rightStrand) {
        if (leftStrand.length() == rightStrand.length()) {
            hamming = IntStream
                    .range(0, leftStrand.length())
                    .reduce(0,
                            (sum, position) -> leftStrand.charAt(position) != rightStrand.charAt(position) ?
                                    ++sum : sum);

        } else {
            if (leftStrand.isEmpty())
                throw new IllegalArgumentException("left strand must not be empty.");
            if (rightStrand.isEmpty())
                throw new IllegalArgumentException("right strand must not be empty.");
            throw new IllegalArgumentException("leftStrand and rightStrand must be of equal length.");
        }

    }

    int getHammingDistance() {
        return hamming;
    }
}

File: kotlin/hamming/src/main/kotlin/Hamming.kt
class Hamming {
    companion object {
        fun compute(initial: String, compare: String): Int {
            return if (initial.length == compare.length) {
                var t = 0
                for (i in 0 until initial.length) {
                    if (initial[i].toUpperCase() != compare[i].toUpperCase()) {
                        t++
                    }
                }
                t
            } else {
                throw  IllegalArgumentException("left and right strands must be of equal length.")
            }
        }
    }
}